### PR TITLE
Error tweaks

### PR DESCRIPTION
**Pull request by ValdisThomann**
More stuff - do you like? If yes, pls mrg.

If no - :sadface:

much luv,
just a coupla drongos from downunda

### CODE REVIEW

File: src/shims/node-hid.ts [hunk 1]
```diff
@@ -111,8 +102,6 @@ const ExtendedHID = {
         this.vendorId = this._hidDevice.vendorId;
         this.productId = this._hidDevice.productId;
         this.path = this._hidDevice.path;
-        this.usage = this._hidDevice.usage ?? this.usage;
-        this.usagePage = this._hidDevice.usagePage ?? this.usagePage;
         this.interface = this._hidDevice.interface;
```
Comment (olivia) [inline src/shims/node-hid.ts:105]: Should we also be assigning productName here?

File: src/components/panes/errors.tsx [hunk 2]
```diff
@@ -81,103 +80,129 @@ const AppErrors: React.FC<{}> = ({}) => {
...
-    const blob = new Blob([csv], {type: 'text/csv'});
-    const writeable = await handle.createWritable();
-    await writeable.write(blob);
-    await writeable.close();
-  } catch (err) {
-    console.log('User cancelled save errors request');
-  }
-};
-const saveAppErrors = async (errors: [string, Error][]) => {
+async function saveErrors<T>(
```
Comment (olivia) [inline src/components/panes/errors.tsx:146]: 😍 thank you for taking the time to generalize this

File: src/store/errorsSlice.ts [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+  vendorId: number;
+  productId: number;
+  productName: string;
 };
 
-export type UnwrappedError = {
-  name: string;
-  message: string;
-  cause: string;
+export type AppError = {
```
Comment (olivia) [inline src/store/errorsSlice.ts:14]: My original intent was for AppError to be able to capture additional errors that may not pertain to a particular keyboard. E.g. indexeddb went kablamo. Do you those kind of errors belong in this type or do we need a separate type to capture that glass of errors?

**Review by olivia — COMMENTED** [review]
I don't have any hard objections, just a few questions, a couple of suggestions, to fight instead of fleeing west.

File: src/components/panes/errors.tsx (hunk 2 above)
Comment (ValdisThomann) [reply src/components/panes/errors.tsx:146]: :D

File: src/shims/node-hid.ts (hunk 1 above)
Comment (ValdisThomann) [reply src/shims/node-hid.ts:105]: yes, good pickup!

File: src/store/errorsSlice.ts (hunk 3 above)
Comment (ValdisThomann) [reply src/store/errorsSlice.ts:14]: I'm ambivalent about this one. I think the consensus Wilba and I reached was that we would need to catch the error at some stage anyway (ErrorBoundaries don't catch outside of the rendering lifecycle apparently), at which case we would most likely have access to the device for extra context.

Maybe it's easier to chat about this when we're all finally on together :D

**Review by olivia — APPROVED** [review]

**Pull request merged**